### ISSUE TITLE
Docs: error in AbstractAuthenticationMiddleware example

### ISSUE THREAD

**Issue by rimai4** [litestar-org/litestar#2305]
### Summary

In the JWTAuthenticationMiddleware example, the following line gives an error:

`engine = cast("AsyncEngine", connection.app.state.postgres_connection)`

calling keys() on `connection.app.state` gives the following result:
`['db_engine', 'session_maker_class']`

**Comment by geeshta** [comment]
I think this entire section needs rewriting. The examples are not even moved to the `examples/` dir.

**Comment by provinzkraut** [comment]
> I think this entire section needs rewriting. The examples are not even moved to the `examples/` dir.

Agreed. If any of you guys want to work on that let me know and I'll assign you (=

**Comment by geeshta** [comment]
@provinzkraut I'll give it a shot.
